src/librustc_resolve/record_exports.rs

```diff
@@ -130,13 +130,14 @@ impl<'a, 'b, 'tcx> ExportRecorder<'a, 'b, 'tcx> {
 
     fn add_exports_for_module(&mut self, exports: &mut Vec<Export>, module_: &Module) {
         for (name, import_resolution) in module_.import_resolutions.borrow().iter() {
-            if !import_resolution.is_public {
-                continue;
-            }
             let xs = [TypeNS, ValueNS];
             for &ns in &xs {
-                match import_resolution.target_for_namespace(ns) {
-                    Some(target) => {
+                if !import_resolution[ns].is_public {
+                    continue;
```

Comment:
No, if the import resolution is not public in one namespace it still could be public in the other namespace. For example, consider
```rust
mod foo {}
pub fn f() {}

use foo as bar;
pub use f as bar;
```
Here, there is a single `import_resolution` corresponding to `bar` that is private in the type namespace (i.e. `!import_resolution[TypeNS].is_public`) but public in the value namespace (i.e. `import_resolution[ValueNS].is_public`). If we skipped to the next import_resolution after seeing that the type namespace is private, we would not export `bar` in the value namespace.